BrainPortal/app/models/message.rb

```diff
@@ -36,6 +36,10 @@ class Message < ApplicationRecord
 
   attr_accessor :send_email
 
+  has_many                :resource_usage
+  before_save             :track_resource_usage_update
```

Comment:
I'd use before_create, because otherwise we are going to get a count every time someone modifies a message.

Reply:
ok